octoprint_mrbeam/gcodegenerator/img2gcode.py

```diff
@@ -455,117 +457,119 @@ def generate_gcode(self, imgArray, xMM, yMM, wMM, hMM, file_id):
                         )
-                        extrema_x = _minmax(self.gc_ctx.x, extrema_next_line)
-                        # start and end are the positions the laserhead should be
-                        # before and after the overshoot.
-                        # Assumes that self.gc_ctx.y was previously set
-                        start = np.array([extrema_x, self.gc_ctx.y])
-                        end = np.array([extrema_x, y])
-                        # /!\ direction_positive reverted at the end of loop
-                        overshoot_gco += self.get_overshoot(
-                            start,
-                            end,
-                            k,
-                            self.overshoot_distance,
-                            offset_counter=row,
-                        )
-                        self._append_gcode(overshoot_gco)
-
-                    # prepare line start
-                    self.write_gcode_for_line_start(
-                        y,
-                        img_pos_mm,
-                        pix,
-                        line_info,
-                        direction_positive,
-                        debug=self.debug,
-                    )
 
-                    # do line
-                    self.write_gcode_for_trimmed_line(
-                        img_pos_mm, pix, line_info, direction_positive, debug=self.debug
-                    )
+                        # do line
+                        self.write_gcode_for_trimmed_line(
+                            img_pos_mm, pix, line_info, direction_positive, debug=self.debug
+                        )
 
-                    # after line
-                    self.write_gcode_for_line_end(
-                        img_pos_mm, line_info, direction_positive, debug=self.debug
-                    )
+                        # after line
+                        self.write_gcode_for_line_end(
+                            img_pos_mm, line_info, direction_positive, debug=self.debug
+                        )
 
-                    # flip direction after each line to go back and forth
-                    direction_positive = not direction_positive
-                    first_row = False
-                elif line_info["left"] != None:
-                    # skip line vertical out of working area
-                    self._append_gcode(
-                        "; ignoring line y={}, out of working area.".format(y)
-                    )
+                        # flip direction after each line to go back and forth
+                        direction_positive = not direction_positive
+                        first_row = False
+                    elif line_info["left"] != None:
+                        # skip line vertical out of working area
+                        self._append_gcode(
+                            "; ignoring line y={}, out of working area.".format(y)
+                        )
 
-            self._append_gcode("; EndPart")
```

Comment:
This signals the end of that design engraving, please move to outside and after the loop